File: relationship-py/relationship/cache.py
"""
缓存数据
"""

from relationship.data.input import input_ as _input
from relationship.data.sort import sort as _sort


def get_cache_data(mode_data: dict[str, list[str]]):
    _hash = {**mode_data}
    for key in _input:
        _hash[key] = _hash.get(key, []) + _input[key]
    for key in _sort:
        _hash[key] = _hash.get(key, []) + _sort[key]

    cache_data: dict[str, list[str]] = {}
    for key in _hash:
        for name in _hash[key]:
            if name not in cache_data:
                cache_data[name] = []
            cache_data[name].append(key)

    return cache_data

File: relationship-go/cache/cache.go
package cache

import (
	"maps"

	"github.com/young2j/relationship/relationship-go/data"
)

func GetCacheData(modeData map[string][]string) map[string][]string {
	_hash := maps.Clone(modeData)
	for sel, inputNames := range data.Input {
		if names, ok := _hash[sel]; ok {
			_hash[sel] = append(names, inputNames...)
		} else {
			_hash[sel] = inputNames
		}
	}

	for sel, sortNames := range data.Sort {
		if names, ok := _hash[sel]; ok {
			_hash[sel] = append(names, sortNames...)
		} else {
			_hash[sel] = sortNames
		}
	}

	cacheData := make(map[string][]string)
	for sel, names := range _hash {
		for _, name := range names {
			if sels, ok := cacheData[name]; ok {
				cacheData[name] = append(sels, sel)
			} else {
				cacheData[name] = []string{sel}
			}
		}
	}

	return cacheData
}
